mono/utils/mono-os-mutex.h

```diff
@@ -151,22 +161,38 @@ mono_os_cond_wait (mono_cond_t *cond, mono_mutex_t *mutex)
 static inline int
 mono_os_cond_timedwait (mono_cond_t *cond, mono_mutex_t *mutex, guint32 timeout_ms)
 {
+	
+#if defined(PLATFORM_MACOSX)
 	struct timeval tv;
-	struct timespec ts;
 	gint64 usecs;
+#endif	
+	struct timespec ts;
 	int res;
-
+	
 	if (timeout_ms == MONO_INFINITE_WAIT) {
 		mono_os_cond_wait (cond, mutex);
 		return 0;
 	}
 
 	/* ms = 10^-3, us = 10^-6, ns = 10^-9 */
-
+#if !defined(PLATFORM_MACOSX)
+/* cond is using CLOCK_MONOTONIC as time source */	
+	res = clock_gettime (CLOCK_MONOTONIC, &ts);
```

Comment:
You shouldn't use the `g_error` from below, as the message is specifically for an error in `gettimeofday`. So please duplicate the error checking code.